### PR TITLE
Sphinx doc chapter 24

### PR DESCRIPTION
**Pull request by maximedenes**
Depends on #7037 and #7041

### CODE REVIEW

File: doc/sphinx/addendum/program.rst [hunk 1]
```diff
@@ -1,329 +1,383 @@
...
+  typed in a context where ∀ p, _ <> S (S p).
++ Coercion. If the object being matched is coercible to an inductive
+  type, the corresponding coercion will be automatically inserted. This
+  also works with the previous mechanism.
+
+
+There are options to control the generation of equalities and
+coercions.
+
+.. cmd:: Unset Program Cases
```
Comment (Zimmi48) [inline doc/sphinx/addendum/program.rst]: Looks like this is not going to appear in the option index.

File: doc/sphinx/addendum/program.rst [hunk 2]
```diff
@@ -1,329 +1,383 @@
...
+
+   This deactivates the special treatment of
+   pattern-matching generating equalities and inequalities when using
+   |Program| (it is on by default). All pattern-matchings and let-patterns
+   are handled using the standard algorithm of |Coq| (see :ref:`extendedpatternmatching`)
+   when this option is deactivated.
+
+.. cmd:: Unset Program Generalized Coercion
```
Comment (Zimmi48) [inline doc/sphinx/addendum/program.rst]: Same remark.

File: doc/sphinx/addendum/program.rst [hunk 3]
```diff
@@ -1,329 +1,383 @@
...
+      .. exn:: In environment … the term: @term does not have type @type. Actually, it has type ...
+
+
+   .. cmdv:: Program Definition @ident @binders : @type := @term.
+
+      This is equivalent to:
+
+      :g:`Program Definition ident : forall binders, type := fun binders => term`.
+
+      .. todo:: refer to production in alias
```
Comment (Zimmi48) [inline doc/sphinx/addendum/program.rst]: Not sure what this TODO means.

File: doc/sphinx/addendum/program.rst [hunk 4]
```diff
@@ -1,329 +1,383 @@
...
+   Admits all obligations (of ``ident``).
+
+   .. note:: Does not work with structurally recursive programs.
+
+.. cmd:: Preterm {? of @ident}
+
+   Shows the term that will be fed to the kernel once the obligations
+   are solved. Useful for debugging.
+
+.. cmd:: Set Transparent Obligations
```
Comment (Zimmi48) [inline doc/sphinx/addendum/program.rst]: Not sure this is going to find its way to the option index like this.

File: doc/sphinx/addendum/program.rst [hunk 5]
```diff
@@ -1,329 +1,383 @@
...
+
+   Control whether all obligations should be declared as transparent
+   (the default), or if the system should infer which obligations can be
+   declared opaque.
+
+.. cmd:: Set Hide Obligations
```
Comment (Zimmi48) [inline doc/sphinx/addendum/program.rst]: Same remark.

**Comment by maximedenes** [comment]
Pipeline at: https://gitlab.com/maximedenes/coq/pipelines/19694976

**Comment by maximedenes** [comment]
@silene Pipeline succeded, can you merge ? (after #7037 and #7041)

**Pull request merged**